### ISSUE TITLE
Connection refused when trying to view Web UI using SSH port forwarding

### ISSUE THREAD

**Issue by stsievert** [dask/distributed#418]
I'm getting started with distributed and running it on a remote cluster, and am trying to verify that it's talking between two different compute machines in a cluster.

Specifically, I run the commands

``` shell
scott@local-machine: ssh -L 8000:localhost:8787 scott@remote-server-1
scott@remote-server-1: dask-scheduler
scott@local-machine: ssh scott@remote-server-2
scott@remote-server-2: dask-worker <ip addr from dask-scheduler>
```

Using this, I can run dask scripts -- I believe it's working, but would like to see. On my local machine I try to access `http://localhost:8000/status` to see the Dask Web UI. When I do this, I see the below error on `remote-machine-1`:

``` bash
channel 3: open failed: connect failed: Connection refused
channel 3: open failed: connect failed: Connection refused
channel 3: open failed: connect failed: Connection refused
```

I've played with it, but I'm not sure where to go next. I would appreciate seeing the [Web UI docs] being expanded on to mention this issue.

[Web UI docs]:http://distributed.readthedocs.io/en/latest/web.html

**Comment by mrocklin** [comment]
> I would appreciate seeing the Web UI docs being expanded on to mention this issue.

I'd be happy to expand the documentation.  But first, what do you think the problem is?  I don't know what your network looks like enough to be able to reasonably solve your problem.  Networks are hard.  Are you able to see other web servers that you host on your network?  How about bokeh server applications?  Are you able to get things running ok on your local machine?

**Comment by stsievert** [comment]
> Are you able to get things running ok on your local machine?

When running on my local machine, I can see the Web UI.

> Are you able to see other web servers that you host on your network? 

With the commands below, I can access a Jupyter notebook on my local machine by going to `http://localhost:8000`.

``` python
scott@local-machine: ssh -L 8000:localhost:8888 scott@remote-server-1
scott@remote-server-1: jupyter notebook
... uses links on remote machine to start jupyter notebook on remote machine ...
```

> what do you think the problem is?

I think the problem is network-related -- somehow SSH tunneling isn't working. I would guess that I need to bind 144.92.142.191:8787 (the IP address given by `dask-scheduler`) to localhost:8787 on the remote machine then SSH in with `ssh -L 8000:localhost:8787`?

**Comment by mrocklin** [comment]
You could try passing the address by which you're calling the server to the bokeh whitelist as follows:

    dask-scheduler --bokeh-whitelist localhost:8000

I don't think this should make a difference, but it's worth a shot.  Other than that I don't really know what's going on.

**Comment by mrocklin** [comment]
Any luck?

**Comment by stsievert** [comment]
I’m going to talk to the IT people that manage this cluster later today. I’ll relay the solution here.




On August 16, 2016 at 10:14:00 AM, Matthew Rocklin (notifications@github.com) wrote:

Any luck?

**Comment by stsievert** [comment]
Huh, well when I went in for help, the example above worked. I'm not sure what changed, but I don't see the error message anymore.

I'd still suggest adding the port forwarding example to the docs.

**Comment by caseyjlaw** [comment]
I've seen these errors sporadically, too. The web UI generally works anyway, so I figured it was some SSH retry feature.

**Comment by mrocklin** [comment]
Fixed by #447
